waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -572,6 +543,44 @@ when defined(rlnzerokit):
     var rootValue = cast[ptr MerkleNode] (root.`ptr`)[]
     return ok(rootValue)
 
+
+proc updateValidRootQueue*(wakuRlnRelay: WakuRLNRelay, root: MerkleNode): void =
+  # updates the valid merkle root queue with the latest root, and pops the oldest one
+  let overflowCount = wakuRlnRelay.validMerkleRoots.len() - AcceptableRootWindowSize
+  if overflowCount >= 0:
+    # Delete the oldest n elements in the deque (index 0..n)
+    for i in 0..overflowCount:
+      wakuRlnRelay.validMerkleRoots.popFirst() 
+  # Push next root into the queue
```

Comment:
```suggestion
  # Push the next root into the queue
```

Reply:
Addressed in [490908d](https://github.com/status-im/nwaku/pull/1177/commits/490908da8efbf14b14d767c4e9d874fad1f94264)